### PR TITLE
Rails 4

### PR DESCRIPTION
**Pull request by ecrisac**
Changes in Tagtical for: https://github.com/Mixbook/product/issues/12411

- Upgrade this gem to allow usage of rails 4
- According to rails 4 philosophy we should not use `attr_accessible` on model layer the validation should be on the controller side. 
https://stackoverflow.com/questions/17371334/how-is-attr-accessible-used-in-rails-4
- Exclude `rubygems_version ` from gemspec, it is autogenerated

From docs:
```Note: If you still want to use attr_accessible, you need to add protected_attributes to your Gemfile. Otherwise, you will be faced with a RuntimeError.```

### CODE REVIEW

File: lib/tagtical/taggable/core.rb [hunk 1]
```diff
@@ -37,7 +37,11 @@ def self.included(base)
 
     module ClassMethods
       def initialize_tagtical_core
-        has_many :taggings, :as => :taggable, :dependent => :destroy, :include => :tag, :class_name => "Tagtical::Tagging"
+        if Rails.version.starts_with?("4")
```
Comment (yuriihohan) [inline lib/tagtical/taggable/core.rb:40]: nitpick - ` Rails::VERSION::MAJOR >= 4`

File: lib/tagtical/taggable.rb [hunk 2]
```diff
@@ -30,8 +30,11 @@ def acts_as_taggable(*tag_types)
         class_attribute :tag_types, :custom_tag_types
         self.tag_types = tag_types
         self.custom_tag_types = tag_types - [Tagtical::Tag::Type::BASE]
-
-        has_many :taggings, :as => :taggable, :dependent => :destroy, :include => :tag, :class_name => "Tagtical::Tagging"
+        if Rails.version.starts_with?("4")
```
Comment (yuriihohan) [inline lib/tagtical/taggable.rb:33]: Rails::VERSION::MAJOR >= 4

File: lib/tagtical/tag.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module Tagtical
   class Tag < ::ActiveRecord::Base
-    attr_accessible :value
+    # attr_accessible :value
```
Comment (yuriihohan) [inline lib/tagtical/tag.rb:3]: can we remove this line?

File: lib/tagtical/tagging.rb [hunk 4]
```diff
@@ -2,14 +2,14 @@ module Tagtical
...
 
-  attr_accessible :tag,
-    :tag_id,
-    :taggable,
-    :taggable_type,
-    :taggable_id,
-    :tagger,
-    :tagger_id,
-    :relevance
+  # attr_accessible :tag,
```
Comment (yuriihohan) [inline lib/tagtical/tagging.rb:5]: can we remove these lines?

**Review by yuriihohan — COMMENTED** [review]
What's the future of this PR?

**Comment by ecrisac** [comment]
> What's the future of this PR?

We should merge this or in com, later

**Review by yuriihohan — APPROVED** [review]
Please add reasons behind the change to the PR description

**Review by yuriihohan — APPROVED** [review]

**Pull request merged**